### PR TITLE
(Bug) Contribution confirmation popup

### LINKED ISSUE #986: (Bug) Contribution confirmation popup
Popup is displaying the amount entered by the user, instead of the actual amount bought

### PR DESCRIPTION
**Pull request by mariano-aguero**
Closes #986 

- Added function to get user token balance
- Check token balance before and after purchase

### CODE REVIEW

**Comment by dennis00010011b** [comment]
@mariano-aguero 
Minted crowdsale: amount of purchase always 0 in popup
Steps:
1. Create Minted, 1 tier, no whitelist
2 Buy some valid amount
3. Observe popup after purchase 

screen shot 2018-07-20 at 01 12 52

File: src/components/contribute/index.js [hunk 1]
```diff
@@ -431,11 +432,19 @@ export class Contribute extends React.Component {
     const { DECIMAL_PLACES } = weiToSend.constructor.config()
     weiToSend.constructor.config({ DECIMAL_PLACES: +tokenStore.decimals })
 
-    const tokensToContribute = weiToSend.div(crowdsalePageStore.rate).toFixed()
     weiToSend.constructor.config({ DECIMAL_PLACES })
 
+    const userBalanceBeforeBuy = await getUserBalance(addr, execID, account)
```
Comment (vbaranov) [inline src/components/contribute/index.js]: @mariano-aguero but we already know, what is the balance of a user before contribution. We display it on the page. Thus, we can get it somewhere from the store. There is no need to get it once more.

Comment (mariano-aguero) [reply src/components/contribute/index.js]: @vbaranov Great.Thanks for the feedback. I added a function that returns the value of the store before purchase and also solve conflicts

**Comment by dennis00010011b** [comment]
@mariano-aguero 
I faced with same issue for Minted crowdsale.
screen shot 2018-07-20 at 08 19 56

**Comment by mariano-aguero** [comment]
I was testing by an hour the problem suggested by @dennis00010011b . Minted and dutch strategy several times, I think it's like a heisenbug. I will continue testing later. 
The video of one test https://www.useloom.com/share/0437a78e2dc14e66adadca8f7e0b8dac

**Comment by dennis00010011b** [comment]
@mariano-aguero 
Testing with ganache
https://www.useloom.com/share/55485b26d6144ea5902e4b039057fe82

**Comment by mariano-aguero** [comment]
@dennis00010011b 
I was able to reproduce the error in my local computer using ganache. 
I add a functionality to perform a retry until the balance is modified in the case of the success of the transaction

**Comment by dennis00010011b** [comment]
@mariano-aguero @vbaranov 
Tested it . No issues were found

**Pull request merged**